src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers.Tests/DiagnosticScopeCodeRefactoringProviderTest.cs

```diff
@@ -0,0 +1,183 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading.Tasks;
+using Xunit;
+using Verifier = Azure.ClientSdk.Analyzers.Tests.AzureRefactoringVerifier<Azure.ClientSdk.Analyzers.DiagnosticScopeCodeRefactoringProvider>;
+
+namespace Azure.ClientSdk.Analyzers.Tests
+{
+    public class DiagnosticScopeCodeRefactoringProviderTest
+    {
+        private static string DiagnosticFramework = @"
+namespace Azure.Core.Pipeline
+{
+    using System;
+    internal readonly struct DiagnosticScope : IDisposable
+    {
+        public void Start() { }
+
+        public void Dispose() { }
+
+        public void Failed(Exception e) { }
+    }
+
+    internal sealed partial class ClientDiagnostics
+    { 
+        public DiagnosticScope CreateScope(string name)
+        {
+            return default;   
+        }
+    }
+}
+";
+
+        [Fact]
+        public async Task AddsScopeToSyncMethod()
+        {
+            const string code = @"
+namespace RandomNamespace
+{
+    using System;
+    using Azure.Core.Pipeline;
+    public class MyClass
+    {
+        ClientDiagnostics _clientDiagnostics;
+        public void [||]Foo(int b)
+        {
+            if (b == 0) throw new ArgumentException(nameof(b));
+            int a = 2 + b;
+        }
+    }
+}";
+            const string fixedCode = @"
+namespace RandomNamespace
+{
+    using System;
+    using Azure.Core.Pipeline;
+    public class MyClass
+    {
+        ClientDiagnostics _clientDiagnostics;
+        public void Foo(int b)
+        {
+            if (b == 0) throw new ArgumentException(nameof(b));
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($""{nameof(MyClass)}.{nameof(Foo)}"");
+            scope.Start();
+            try
+            {
+                int a = 2 + b;
+            }
+            catch (Exception ex)
+            {
+                scope.Failed(ex);
+                throw;
+            }
+        }
+    }
+}";
+
+            await Verifier.CreateRefactoring(code, fixedCode)
+                .WithSources(DiagnosticFramework)
+                .RunAsync();
+        }
+
+        [Fact]
+        public async Task AddsScopeToExpressionSyncMethod()
+        {
+            const string code = @"
+namespace RandomNamespace
+{
+    using System;
+    using Azure.Core.Pipeline;
```

Comment:
This is not the best pattern to use for using statements. Any reason not to have them outside of the namespace declaration? Doesn't make a difference here.

Reply:
I think I copied it from somewhere. Doesn't matter much here but pretty easy to fix.